### PR TITLE
Make sure to support 8.1

### PR DESCRIPTION
**Pull request by harikt**
Trying to address #194 , #208 .

Some credits for #204

### CODE REVIEW

**Comment by kenjis** [comment]
I ran tests on PHP 8.1. Is it okay?

```
$ vendor/bin/phpunit 
PHPUnit 9.5.16 by Sebastian Bergmann and contributors.

  Warning - The configuration file did not pass validation!
  The following problems have been detected:

  Line 3:
  - Element 'testsuite': The attribute 'name' is required but missing.

  Line 7:
  - Element 'filter': This element is not expected.

  Test results may not be as expected.
```

**Review by koriym — APPROVED** [review]

**Comment by harikt** [comment]
Hey @kenjis , 

That was phpunit configuration issue. I have changed it now. The tests are running on 8.1, you can see in github actions.

**Comment by kenjis** [comment]
@harikt Okay, I've confirmed the warnings are gone.

**Review by kenjis — APPROVED** [review]

File: tests/ExtendedPdoTest.php [hunk 1]
```diff
@@ -105,6 +106,9 @@ public function testConnectionQueries()
 
         // make sure new encoding was honored
         $actual = $pdo->fetchValue('PRAGMA encoding');
+
+        // When changing to UTF-16, sometimes it is returning UTF-16le
+        // $this->assertEquals($newEncoding, $actual);
```
Comment (harikt) [inline tests/ExtendedPdoTest.php:111]: Can anyone of you run the test uncommenting the line ?

This is what I get

image

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+    <include>
+      <directory suffix=".php">./src</directory>
+    </include>
+  </coverage>
+  <testsuites>
+    <testsuite name="Aura.Sql test suite">
+      <directory>./tests</directory>
+    </testsuite>
+  </testsuites>
+</phpunit>
```
Comment (koriym) [inline phpunit.xml.dist:13]: missing EOL?

Comment (harikt) [reply phpunit.xml.dist:13]: Is that necessary?

**Comment by harikt** [comment]
Someone thought I am disrespectful to them. https://github.com/YOURLS/YOURLS/pull/3230#issuecomment-1052095127 . I am not sure if I did something wrong helping the project.

Let me step back.

File: tests/ExtendedPdoTest.php (hunk 1 above)
Comment (kenjis) [reply tests/ExtendedPdoTest.php:111]:
```
$ vendor/bin/phpunit tests/ExtendedPdoTest.php
PHPUnit 9.5.16 by Sebastian Bergmann and contributors.

.F............................................                    46 / 46 (100%)

Time: 00:00.090, Memory: 6.00 MB

There was 1 failure:

1) Aura\Sql\ExtendedPdoTest::testConnectionQueries
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'UTF-16'
+'UTF-16le'
```

**Comment by kenjis** [comment]
It seems they were mad because they thought their PR was closed without any explanation.

File: phpunit.xml.dist (hunk 2 above)
Comment (koriym) [reply phpunit.xml.dist:13]: maybe https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline or not?

**Comment by LeoColomb** [comment]
Yes, I'm not mad about the PR itself, but more about the way it has been managed.
Might be a misunderstanding, if so I'm sorry about that.

Yes indeed, my PR has been closed without any explanation, and yes this is contributor harmful.
Usually I don't care, but as this PR was requested, it becomes very touchy.

Also @harikt, be careful with who helps who. We never asked for any help on the mentioned project, YOURLS.
You have commented a change in YOURLS to gather details (which is perfectly fine, glad to help), then _you_ asked for help on _your_ project, Aura.Sql. Not the same!

**Comment by harikt** [comment]
Hey @LeoColomb ,

Firstly and lastly I apologize if you felt I was rude. Normally I used to wait for the changes so that people get credits for what they deserve.
 
Today I noticed a closed PR. I thought this was closed by you. Again I didn't closed it myself. If so it may have happened by an accident.

If you got an email stating harikt closed the issue, can you post a screenshot?

I am also curious to know what really happened.

**Comment by kenjis** [comment]
Normally it shows who and when closed:
Screenshot 2022-02-27 7 52 55

But the PR's closed comment is different:
Screenshot 2022-02-27 7 52 42

**Comment by LeoColomb** [comment]
Indeed, I didn't received any email about the closure and more importantly my fork does not exist anymore for no reason.
Such an embarrassing situation...
I've sent GitHub support a message to request an investigation.

So, should I recreate my PR?

**Comment by harikt** [comment]
> So, should I recreate my PR?

No. This has been addressed currently.

Thank you for your time.

**Comment by koriym** [comment]
I was just about to request GitHub support too!

**Comment by pmjones** [comment]
This is very weird.

@LeoColomb I know how it is when someone has been rude on the internet -- it makes you feel frustrated and angry and you want to push back in return. But if you will take my word for it, @harikt is one of the most polite and respectful people I know, so any sense of rudeness was sure to be a miscommunication.

Thank you @LeoColomb for your patience and understanding.

**Comment by koriym** [comment]
Support responded.

> Pull requests can sometimes close automatically due to related events, rather than being closed directly by a user. For example, if the pull request originates from a fork, and the fork is then deleted, the associated pull request will be closed automatically. This is because the branch in the pull request no longer exists, having been deleted as part of the fork.

It's not like any person closed the PR directly.

**Comment by harikt** [comment]
Thank you @pmjones , @koriym , @kenjis  for taking your valuable time and trying to resolve the misunderstanding happened due to an event with github. @LeoColomb lets forget the bad stuffs and move on.

Have a nice day ahead.

With love

Hari

**Pull request merged**